mayastor/src/nexus_uri.rs

```diff
@@ -76,3 +82,42 @@ pub async fn bdev_destroy(uri: &str) -> Result<(), NexusBdevError> {
 pub fn bdev_get_name(uri: &str) -> Result<String, NexusBdevError> {
     Ok(Uri::parse(uri)?.get_name())
 }
+
+impl std::cmp::PartialEq<url::Url> for &Bdev {
+    fn eq(&self, uri: &url::Url) -> bool {
+        match Uri::parse(&uri.to_string()) {
+            Ok(device) => {
+                let uri_driver = if uri.scheme() == "nvmf" {
+                    "nvme"
+                } else {
+                    uri.scheme()
+                };
+                uri_driver == self.driver() && device.get_name() == self.name()
+            }
+            _ => false,
+        }
+    }
+}
+
+impl TryFrom<&Bdev> for url::Url {
```

Comment:
This is nice .. however..... the From<T> and TryFrom<T> typically take a T not &T. This alsi imlies that calling TryFrom<T> consumes T. This is not a hard rule but according to the rust docs, what is supposed to happen. so when calling this, what about, if we we do try_from(bdev.clone())?

Reply:
Yeah that would not be ideal indeed, I'll have a look!

Reply:
hmm [here](https://doc.rust-lang.org/std/ops/index.html) seems to say that we should add &T|T as required?
Regardless, T with clone does work as well so we can go around consuming T with that.